### ISSUE TITLE
BUG: Inconsistency with grouping columns in agg

### ISSUE THREAD

**Issue by rhshadrach** [pandas-dev/pandas#50383]
Related to #46944

Users can currently request the grouping column be part of the computation for various ops by including them as part of a `__getitem__`. But `agg` will still exclude these columns.

```
df = pd.DataFrame({'a': [1, 1, 2], 'b': 3, 'c': 4, 'd': 5})
gb = df.groupby(['a', 'b'])[['a', 'c']]

result = gb.sum()
print(result)
#      a  c
# a b      
# 1 3  2  8
# 2 3  2  4

result2 = gb.agg(lambda x: x.sum())
print(result2)
#      c
# a b   
# 1 3  8
# 2 3  4
```

I would expect `__getitem__` to only subset columns for groupby rather than being able to add additional (grouping) columns.

**Comment by ihsansecer** [comment]
Working fine on main now
```
>>> result2 = gb.agg(lambda x: x.sum())
>>> print(result2)
     a  c
a b      
1 3  2  8
2 3  2  4
```

**Comment by rhshadrach** [comment]
Thanks @ihsansecer! Would you have any interest in adding a test?

> I would expect `__getitem__` to only subset columns for groupby rather than being able to add additional (grouping) columns.

I've come around to thinking of this as a feature. While groupby typically excludes the grouping column, if there is a use case for including them I suppose it can be good to have.

**Comment by ihsansecer** [comment]
> I've come around to thinking of this as a feature. While groupby typically excludes the grouping column, if there is a use case for including them I suppose it can be good to have.

Yeah I would expect the same and I wasn't aware of this behavior before. Added a test for this
